source/MaterialXRenderGlsl/GlslProgram.cpp

```diff
@@ -626,6 +627,96 @@ void GlslProgram::bindTextures(ImageHandlerPtr imageHandler)
     }
 }
 
+void GlslProgram::bindColorManagement(ColorManagementSystemPtr cms, ImageHandlerPtr imageHandler)
+{
+    if (!cms || !imageHandler)
+    {
+        return;
+    }
+
+    // Handle non-texture uniforms
+    ColorManagementResourceMapPtr uniformItems = cms->getResource(ColorManagementSystem::ResourceType::UNIFORM);
+    if (uniformItems)
+    {
+        for (auto uniformItem : *uniformItems)
+        {
+            std::string uniformName = uniformItem.first;
+            ColorSpaceConstantPtr uniformPtr = std::static_pointer_cast<ColorSpaceConstant>(uniformItem.second);
+            ValuePtr uniformValue = uniformPtr->_value;
+            if (hasUniform(uniformName))
+            {
+                bindUniform(uniformName, uniformValue);
+            }
+        }
+    }
+
+    // Handle texture uniforms. 
+    // Note: Need support for 3D images to support binding here
+    ColorManagementSystem::ResourceType textureTypes[] = {
+        ColorManagementSystem::ResourceType::TEXTURE1D, ColorManagementSystem::ResourceType::TEXTURE2D
+        /* , ColorManagementSystem::ResourceType::TEXTURE3D */
+    };
+
+    for (auto textureType : textureTypes)
+    {
+        uniformItems = cms->getResource(textureType);
+        if (uniformItems)
+        {
+            ImageSamplingProperties samplingProperties;
+            samplingProperties.uaddressMode = ImageSamplingProperties::AddressMode::CLAMP;
+            samplingProperties.vaddressMode = ImageSamplingProperties::AddressMode::CLAMP;
+            samplingProperties.filterType = ImageSamplingProperties::FilterType::LINEAR;
+
+            for (auto uniformItem : *uniformItems)
+            {
+                std::string uniformName = uniformItem.first;
+                ColorSpaceTexturePtr uniformTexture = std::static_pointer_cast<ColorSpaceTexture>(uniformItem.second);
+
+                // Create an new image if needed and bind it to the appropriate location
+                // Note that if the image with the uniform name already exists then it will be
+                // returned from the image cache and reused.
+                if (hasUniform(uniformName))
+                {
+                    ImagePtr uniformImage = imageHandler->getImage(uniformName);
+                    if (!uniformImage)
+                    {
+                        uniformImage = imageHandler->createImage(uniformName,
```

Comment:
Create image as needed and bind. As noted in the PR the Image type does not suport 3d images so this will need to be added later on.